packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ClientProvider.cs

```diff
@@ -31,18 +63,159 @@ public ClientProvider(InputClient inputClient)
+
+            // add other parameter assignments to their corresponding fields
+            foreach (var p in primaryConstructorParameters)
+            {
+                if (p.Field != null)
+                {
+                    body.Add(p.Field.Assign(p).Terminate());
+                }
+            }
+
+            // handle pipeline field
+            ValueExpression perRetryPolicies = New.Array(typeof(PipelinePolicy));
+            if (_authorizationHeaderConstant != null && _apiKeyAuthField != null)
+            {
+                // new PipelinePolicy[] { ApiKeyAuthenticationPolicy.CreateHeaderApiKeyPolicy(_keyCredential, AuthorizationHeader) }
+                ValueExpression[] perRetryPolicyArgs = _authorizationApiKeyPrefixConstant != null
+                    ? [_apiKeyAuthField, _authorizationHeaderConstant, _authorizationApiKeyPrefixConstant]
+                    : [_apiKeyAuthField, _authorizationHeaderConstant];
+                var perRetryPolicy = Static<ApiKeyAuthenticationPolicy>().Invoke(
+                    nameof(ApiKeyAuthenticationPolicy.CreateHeaderApiKeyPolicy), perRetryPolicyArgs).As<ApiKeyAuthenticationPolicy>();
+                perRetryPolicies = New.Array(typeof(PipelinePolicy), isInline: true, perRetryPolicy);
+            }
+
+            body.Add(PipelineField.Assign(ClientPipelineSnippets.Create(
+                _clientOptionsParameter, New.Array(typeof(PipelinePolicy)), perRetryPolicies, New.Array(typeof(PipelinePolicy)))).Terminate());
+
+            // TO-DO: Add additional field assignments for client options https://github.com/microsoft/typespec/issues/3688
+
+            return [.. body];
+        }
+
+        /// <summary>
+        /// Builds the secondary constructor for the client. The secondary constructor contains all required parameters as arguments.
+        /// </summary>
+        /// <param name="requiredParams">The required parameters for the client.</param>
+        /// <param name="primaryCtorOrderedParams">The ordered parameters for the primary constructor.</param>
+        private ConstructorProvider BuildSecondaryConstructor(
+            IReadOnlyList<ParameterProvider> requiredParams,
+            IReadOnlyList<ParameterProvider> primaryCtorOrderedParams)
+        {
+            // initialize the arguments for the primary constructor
+            var primaryCtorArgs = new List<ValueExpression>(primaryCtorOrderedParams.Count);
+            foreach (var p in primaryCtorOrderedParams)
+            {
+                if (p.InitializationValue == null && p.Validation != ParameterValidationType.None)
+                {
+                    primaryCtorArgs.Add(p);
+                }
+                else
+                {
+                    primaryCtorArgs.Add(p.InitializationValue ?? DefaultOf(p.Type));
+                }
+            }
+
+            var primaryCtorInitializer = new ConstructorInitializer(false, primaryCtorArgs);
```

Comment:
I think you can simplify the list and avoid the allocation of primaryCtorArgs by doing this

```c#
var primaryCtorInitializer = new ConstructorInitializer(false, [.. primaryCtorOrderedParams.Select(p => p.InitializationValue ?? DefaultOf(p.Type))]);
```